pulp_rpm/app/serializers.py

```diff
@@ -131,8 +164,32 @@ class PackageSerializer(ContentSerializer):
                     "file mtime in seconds since the epoch.")
     )
 
+    def create(self, validated_data):
+        """
+        Create a Package.
+
+        Overriding default create() to deal with artifact properly.
+
+        Args:
+            validated_data (dict): Data used to create the Package
+
+        Returns:
+            models.Package: The created Package
+
+        """
+        artifact = validated_data.pop('artifact')
+
+        package = Package.objects.create(**validated_data)
+        ca = ContentArtifact(artifact=artifact,
+                             content=package,
+                             relative_path=package.filename)
+        ca.save()
+
+        return package
+
     class Meta:
-        fields = ContentSerializer.Meta.fields + (
+        fields = tuple(set(ContentSerializer.Meta.fields) - {'artifacts'}) + (
+            'artifact',
```

Comment:
this is interesting, it subtracts 'artifacts' and then adds it back as a singular? Is ContentSerializer incorrect to include 'artifacts' here?

Reply:
This is what we're doing in Pulp File and Pulp Python.  The default assumption is that there can be more than one artifact per content unit, but for these content types that isn't the case, so it's made more explicit.  I'm not sure what the correct default is -- it's easier to hack it this way than the reverse, but most content types are likely single-artifact.

Reply:
That's ok let's leave this as is at least for now. Thank you for explaining it to me. I think to improve it we would need to have core offer a No, Single, Many Content unit parent classes.